### ISSUE TITLE
t.run is not a function

### ISSUE THREAD

**Issue by SiAdcock** [Jameskmonger/modunit#5]
I have written a simple test:

```
const test = require('modunit');

test('all about the ones', function(t) {
  const one = 1;
  const anotherOne = 1;
  t.assert.equal(one, anotherOne);
});
```

Installed `modunit` and `tap-diff` as `devDependencies`.

I run the following command:

```
$ node spec/my.first.test.js | node node_modules/.bin/tap-diff
```

And I get the following error:

```
/Users/simonadcock/workspace/noths/breadboard/node_modules/modunit/lib/session.js:42
                t.run();
                  ^

TypeError: t.run is not a function
    at Session.<anonymous> (/Users/simonadcock/workspace/noths/breadboard/node_modules/modunit/lib/session.js:42:19)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

I'm using node v4.3.2

Any idea what may be going wrong?

**Comment by Jameskmonger** [comment]
Hi @siadcock, what version of modunit are you using?

**Comment by SiAdcock** [comment]
v0.1.3

**Comment by Jameskmonger** [comment]
Thanks Simon I'll look into this.

**Comment by Jameskmonger** [comment]
Hi @SiAdcock.. not 100% sure what's going on here. 

I've done the following steps, but I'm not able to reproduce the problem:

1. install `modunit@0.1.3`
2. install `tap-diff@0.1.1`
3. create a file `test.js` with the following contents:
  ```javascript
  const test = require('modunit');

  test('all about the ones', function(t) {
    const one = 1;
    const anotherOne = 1;
    t.assert.equal(one, anotherOne);
  });
  ```
4. Run the following command: `node test.js | node node_modules/.bin/tap-diff`

I get this output:

```
james@JM-LAPTOP:~/Development/m/modunit-test$ node test.js | node node_modules/.bin/tap-diff 

  all about the ones
    ✔  equal: [1, 1]

passed: 1  failed: 0  of 1 tests  (29ms)

All of 1 tests passed!
```

**Comment by SiAdcock** [comment]
Hi James

I followed those exact steps and still get this output:

```
[11:19:56] simonadcock:modunit-test $ node test.js | node node_modules/.bin/tap-diff
/Users/simonadcock/workspace/modunit-test/node_modules/modunit/lib/session.js:42
                t.run();
                  ^

TypeError: t.run is not a function
    at Session.<anonymous> (/Users/simonadcock/workspace/modunit-test/node_modules/modunit/lib/session.js:42:19)
    at processImmediate [as _immediateCallback] (timers.js:383:17)

passed: 0  failed: 0  of 0 tests  (8ms)

All of 0 tests passed!
```

Possibly an issue with running this on *nix OS?

**Comment by SiAdcock** [comment]
Okay, I have found a fix and it's related to the inheritance tree.

Currently in simple-test.js, you have:

```js
    function SimpleTest () {
      //constructor things
    }
    SimpleTest.prototype.run = function () {
        this.emit('beforeRun');
        this._callback();
    };
    __inherits(SimpleTest, EventEmitter);
```

`inherits` breaks the link to `SimpleTest`'s prototype, so the `run` method is no longer available to `SimpleTest`.

This also happens in other places, such as assert.js

Calling `inherits` before attaching the `run` method to the prototype fixes this particular issue, but I'm not sure right now if this is the correct solution.

Let me know if this makes sense

**Comment by Jameskmonger** [comment]
Well this is odd! I ran my test on Ubuntu 14.04 and it works fine. I'm not able to replicate this issue at all whether I pipe into tap-diff or not.

Would you be able to zip your test folder up, along with node_modules, and upload it to Google Drive or similar?

I'd be interested to see if I could recreate it that way @SiAdcock

**Comment by Jameskmonger** [comment]
Fixed by #6
